src/casts/interpret-casts-common.rkt

```diff
@@ -163,7 +163,7 @@ TODO write unit tests
        [(type-dyn?$  t2) t1]
        [(and$ (type-fn?$ t1) (type-fn?$ t2)
               (op=? (type-fn-arity$ t1) (type-fn-arity$ t2)))
-        (Make-Fn-Type tglb-uid t1 t2)]
+        (Make-GLB-Two-Fn-Types tglb-uid t1 t2)]
```

Comment:
Another question might be whether or not this should be implemented at this level. For instance I lifted the compose operation to both of these levels instead of in specify-representation. With the addition of assignments we can make both of these operations iterative.

Reply:
I agree on making them iterative and implementing them using higher-level abstractions earlier. It looks ugly/unreadable now and does not really benefit from the low level operations here. However, I would prefer to it do it in another PR that does not focus on `specify-representation` only.